### PR TITLE
fix: Removes need for docker and nodemon

### PR DESCRIPTION
**Pull request by kristianeboe**
Removes
- nodemon from package.json
- all use of prisma and docker
- the entire depreciated wallet provider
- the /api/wallets api routes
- the /archive/home (old home page)

You can still
- edit old stores
- presumably create new ones
Only from the footer though

I tested it quite a bit on localhost just clicking around everywhere, both signed in and not, so I think it's safe :)

### CODE REVIEW

File: pages/marketplace/new.tsx [hunk 1]
```diff
@@ -74,7 +73,10 @@ export default function New() {
   const [form] = Form.useForm();
   const [pendingAddress, setPendingAddress] = useState<string>();
   const { setVisible } = useWalletModal();
-  const { solana, wallet, looking } = useContext(WalletContext);
+  const solana = useWallet();
```
Comment (austinmilt) [inline pages/marketplace/new.tsx:76]: This variable name is pretty confusing IMO. Better would be `userWallet`

File: pages/marketplace/new.tsx [hunk 2]
```diff
@@ -74,7 +73,10 @@ export default function New() {
...
+  const { publicKey } = solana;
+  const pubkey = publicKey?.toBase58();
```
Comment (austinmilt) [inline pages/marketplace/new.tsx:78]: similarly, `pubkey` is pretty ambiguous IMO. Better would be like `userWalletPubKey` or something

File: src/lib/utils/index.ts [hunk 3]
```diff
@@ -15,8 +15,8 @@ export const maybeCDN = (uri: string) => {
 export const getEnvironmentVariable = (environmentVariable: string): string => {
   const unvalidatedEnvironmentVariable = process.env[environmentVariable];
   if (!unvalidatedEnvironmentVariable) {
-    return '';
     console.error(`Couldn't find env var: ${environmentVariable}`);
+    return '';
```
Comment (austinmilt) [inline src/lib/utils/index.ts:19]: IME environment variables should be considered prerequisites for startup/functioning, so a missing environment variable should lead to complete failure, or at least throw an error that must be handled by the caller.

File: src/modules/storefront/provider.tsx [hunk 4]
```diff
@@ -1,35 +1,27 @@
...
-  storefront?: Storefront;
-};
-
-type StorefrontProviderProps = {
-  wallet?: Wallet;
-  children: (props: StorefrontProviderChildrenProps) => React.ReactElement;
-};
-
-export const StorefrontProvider = ({ wallet, children }: StorefrontProviderProps) => {
+export const StorefrontProvider: FC = ({ children }) => {
```
Comment (austinmilt) [inline src/modules/storefront/provider.tsx:9]: good opportunity to switch to our new pattern

```suggestion
export function StorefrontProvider(props: {children: React.ReactElement}): JSX.Element {
```

File: src/modules/storefront/provider.tsx [hunk 5]
```diff
@@ -39,11 +31,11 @@ export const StorefrontProvider = ({ wallet, children }: StorefrontProviderProps
       setStorefront(storefront);
       setSearching(false);
     });
-  }, [wallet?.pubkey]);
+  }, [pubkey]);
 
   return (
     <StorefrontContext.Provider value={{ searching, storefront }}>
-      {children({ searching, storefront })}
+      {children}
```
Comment (austinmilt) [inline src/modules/storefront/provider.tsx:38]: If taking above suggestion, then:

```suggestion
      {props.children}
```

File: src/modules/utils/image.ts [hunk 6]
```diff
@@ -29,19 +26,16 @@ function tryGetImageFromPublicKey(publicKey?: string | PublicKey | null): string
   return result;
 }
 
-
 export function getFallbackImage(seed?: number | string | PublicKey | null): string {
   let seedNumber: number = 1;
-  if ((seed !== null) && (seed != undefined)) {
+  if (seed !== null && seed != undefined) {
```
Comment (austinmilt) [inline src/modules/utils/image.ts:31]: :(

File: src/views/home/Footer.tsx [hunk 7]
```diff
@@ -55,11 +55,11 @@ const Footer: VFC = () => {
               <li>
                 <InternalLink href="/nfts/new">Mint NFTs</InternalLink>
               </li>
-              <li>
+              {/* <li>
```
Comment (austinmilt) [inline src/views/home/Footer.tsx:58]: intentional?

**Review by austinmilt — CHANGES_REQUESTED** [review]
Mostly small stuff. I do think missing environment variables should throw an error

File: src/views/home/Footer.tsx (hunk 7 above)
Comment (kespinola) [reply src/views/home/Footer.tsx:58]: Yeah this isn't active yet so good to hide.

**Review by kespinola — APPROVED** [review]
LG2M. Largest impact I see is on storefront create and edit if you want to give that a test run. Thank you. @austinmilt has good hygiene comments.

File: src/lib/utils/index.ts (hunk 3 above)
Comment (kristianeboe) [reply src/lib/utils/index.ts:19]: This is some crossmint stuff Milo made, he can fix it later :)

Comment (mpwsh) [reply src/lib/utils/index.ts]: my fault here lmao, sorry. it was more like a debugging thing :D

**Pull request merged**